src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,235 @@
+            ITry<string, NetworkOperationError> stringifiedNumber = downloadedNumber.Map(r => r.ToString());
+        }
+
+        private void TransformingErrorValuesWithMapError()
+        {
+            ITry<int, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => DownloadNumberOverNetwork());
+
+            // You can use MapError to map for example from exception to another type you use to represent errors. Or map to logging messages.
+            ITry<int, string> flooredMultiplicationResult = multiplicationResult.MapError(e => e.ToString());
+        }
+
+        private void HandlingNestedTriesWithFlatMap()
+        {
+            ITry<int, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => DownloadNumberOverNetwork());
+            ITry<ITry<int, NetworkOperationError>, NetworkOperationError> resultOfDoubleMultiplication = multiplicationResult.Map(r => PerformNetworkOperation(_ => TransformNumberOverNetwork(r)));
+
+            // This try succeeds only if both tries succeed. However the second lambda is only executed in case the first try is successful.
+            ITry<int, NetworkOperationError> flattenedResultOfDoubleMultiplication = multiplicationResult.FlatMap(r => PerformNetworkOperation(_ => TransformNumberOverNetwork(r)));
+        }
+
+        private void AggregatingMultipleTriesIntoSingleResult(string name, string age, string heightInCentimeters)
+        {
+            // You need to start with tries that have a collection as the type of error. So we MapError to a collection of strings here. But it can be a collection of any type.
+            // For validations, you would write your validation method, so it already returns IEnumerable as the error type and use MapError inside the method.
+            // Try.Aggregate method always executes all tries and then aggregates the success results or the errors. It doesn't stop on first error.
+            // You can transform the result into any class you want. This makes it nice and effective way of parsing values and combining the results.
+            ITry<string, IEnumerable<PersonParsingError>> parsedName = name.ToNonEmptyOption().ToTry(_ => new [] { PersonParsingError.NameNotProvided });
+
+            // Notice that you can chain validations using FlatMap and then pass it into Aggregate.
+            var ageAsValidNumber = Try.Catch<int, Exception>(_ => Convert.ToInt32(age)).MapError(_ => PersonParsingError.AgeNotANumber);
+            var notNegativeAge = ageAsValidNumber.FlatMap(
+                a => a.ToOption().Where(aa => aa >= 0).ToTry(_ => PersonParsingError.AgeNegative)
+            );
+            var validAge = notNegativeAge.FlatMap(
+                a => a.ToOption().Where(aa => aa < 140).ToTry(_ => PersonParsingError.AgeTooHigh)
+            );
+            ITry<int, IEnumerable<PersonParsingError>> parsedAge = validAge.MapError(e => new [] { e });
```

Comment:
This may look weird if you see the example for the first time. Even though it is more code, I'd construct `ITry<string, IEnumerable<PersonParsingError>>` right away, just like you do with `parsedName`. I think that would make it easier to understand.